File: server/src/types/models/src/lead.model.ts
import type {
  CaseType,
  UrgencyLevel,
  LeadStatus,
  ContactAvailability,
} from '@enums';

export interface Lead {
  id: string;
  whatsappNumber: string;
  name: string | null;
  email: string | null;
  cpf: string | null;
  caseType: CaseType | null;
  caseDescription: string | null;
  urgency: UrgencyLevel | null;
  contactAvailability: ContactAvailability | null;
  status: LeadStatus;
  convertedUserId: string | null;
  assignedLawyerId: string | null;
  lawyerNotes: string | null;
  discardReason: string | null;
  isAIPaused: boolean;
  createdAt?: Date;
  updatedAt?: Date;
}

File: mobile/lib/features/lawyer/leads/data/models/lead_model.dart
import '../../domain/entities/lead.dart';

final class LeadModel extends Lead {
  const LeadModel({
    required super.id,
    required super.whatsappNumber,
    required super.status,
    super.name,
    super.cpf,
    super.caseType,
    super.caseDescription,
    super.urgency,
    super.contactAvailability,
    super.createdAt,
  });

  factory LeadModel.fromJson(Map<String, dynamic> json) {
    return LeadModel(
      id: json['id'] as String,
      whatsappNumber: json['whatsappNumber'] as String? ?? '',
      name: json['name'] as String?,
      cpf: json['cpf'] as String?,
      caseType: json['caseType'] as String?,
      caseDescription: json['caseDescription'] as String?,
      urgency: json['urgency'] as String?,
      contactAvailability: json['contactAvailability'] as String?,
      status: json['status'] as String? ?? 'PENDING',
      createdAt: _date(json['createdAt']),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'whatsappNumber': whatsappNumber,
      'name': name,
      'cpf': cpf,
      'caseType': caseType,
      'caseDescription': caseDescription,
      'urgency': urgency,
      'contactAvailability': contactAvailability,
      'status': status,
      'createdAt': createdAt?.toIso8601String(),
    };
  }
}

DateTime? _date(Object? value) {
  if (value == null) return null;
  return DateTime.tryParse(value.toString());
}
